# Z-Test for Population Mean

This notebook demonstrates how to perform a one-sample Z-test to determine if there's a statistically significant difference between a sample mean and a known population mean.

## Import Required Libraries

We need numpy for numerical operations and scipy.stats for the normal distribution functions.

In [ ]:
# Import required libraries for statistical analysis
import numpy as np  # For numerical operations and array manipulation
from scipy.stats import norm  # For normal distribution functions and statistical tests

## Define Sample and Population Parameters

Here we set up our test scenario:
- **Sample**: A collection of height measurements (in cm) from 10 individuals
- **Population mean**: The known average height (170 cm) for the larger population
- **Population standard deviation**: The known variability (3 cm) in the population
- **Sample mean**: The average height of our sample group
- **Sample size (n)**: Number of observations in our sample

This represents a scenario where we want to test if our sample group's average height differs significantly from the general population.

In [ ]:
# Define our test data and population parameters
sample = [172, 174, 168, 169, 171, 173, 175, 170, 169, 172]  # Sample height measurements in cm
population_mean = 170  # Known population average height (cm)
population_std = 3  # Known population standard deviation (cm)
samp_mean = np.mean(sample)  # Calculate sample mean
n = len(sample)  # Get sample size

## Calculate the Z-Score

The Z-score measures how many standard deviations our sample mean is away from the population mean. The formula is:

**Z = (Sample Mean - Population Mean) / (Population Standard Deviation / √Sample Size)**

The denominator represents the standard error of the mean, which accounts for the fact that we're working with a sample rather than the entire population.

In [ ]:
# Calculate the Z-score using the formula: Z = (sample_mean - population_mean) / (population_std / sqrt(n))
z_score = (samp_mean - population_mean) / (population_std / np.sqrt(n))
z_score  # Display the calculated Z-score

np.float64(1.3703203194063098)

## Calculate the P-Value

The p-value represents the probability of observing a test statistic as extreme as (or more extreme than) our calculated Z-score, assuming the null hypothesis is true.

For a two-tailed test (testing if the mean is different in either direction), we:
1. Take the absolute value of the Z-score
2. Find the cumulative probability up to that value using the normal distribution
3. Subtract from 1 to get the upper tail probability
4. Multiply by 2 to account for both tails

A small p-value (typically < 0.05) suggests strong evidence against the null hypothesis.

In [ ]:
# Calculate p-value for two-tailed test: p = 2 * (1 - CDF(|Z-score|))
p_value = 2 * (1 - norm.cdf(abs(z_score)))  # Multiply by 2 for two-tailed test

## Display the P-Value

Let's view the calculated p-value to see the strength of evidence against our null hypothesis.

In [ ]:
# Display the calculated p-value
p_value  # Shows probability of observing such a difference by chance

np.float64(0.17058693287143756)

## Make Statistical Decision

Now we compare our p-value to the significance level (α = 0.05) to make a decision:

- **If p-value < α**: Reject the null hypothesis (evidence suggests the sample mean differs significantly from population mean)
- **If p-value ≥ α**: Fail to reject the null hypothesis (insufficient evidence to claim a significant difference)

The significance level of 0.05 means we're willing to accept a 5% chance of making a Type I error (rejecting a true null hypothesis).

In [ ]:
# Make statistical decision based on p-value comparison with significance level
alpha = 0.05  # Set significance level (5% chance of Type I error)

if p_value < alpha:
    print("I will reject the null hypothesis")  # Strong evidence against H0
else:
    print("i will accept the null hypothesis ")  # Insufficient evidence to reject H0

i will accept the null hypothesis 


## Interpretation of Results

Based on our analysis:
- **Z-score**: 1.37 (moderate distance from population mean)
- **P-value**: 0.171 (17.1% probability of observing such a difference by chance)
- **Decision**: Fail to reject the null hypothesis

**Conclusion**: There is insufficient statistical evidence to conclude that our sample's mean height differs significantly from the population mean of 170 cm. The observed difference could reasonably occur due to random sampling variation.

## Key Concepts Summary

1. **Null Hypothesis (H₀)**: The sample mean equals the population mean
2. **Alternative Hypothesis (H₁)**: The sample mean differs from the population mean
3. **Z-test**: Used when population standard deviation is known and sample size is reasonable
4. **Two-tailed test**: Tests for differences in either direction
5. **Statistical significance**: Determined by comparing p-value to significance level

In [ ]:
# NOTE: The following cells are duplicates and have been removed for clarity
# The complete Z-test analysis is shown above with comprehensive explanations

In [9]:
z_score = (samp_mean - population_mean) / (population_std / np.sqrt(n))
z_score

np.float64(1.3703203194063098)

In [10]:
p_value = 2 * (1 - norm.cdf(abs(z_score)))

In [11]:
p_value

np.float64(0.17058693287143756)

In [12]:
alpha = 0.05

if p_value < alpha:
    print("I will reject the null hypothesis")
else:
    print("i will accept the null hypothesis ")

i will accept the null hypothesis 
